xrpl4j-model/src/main/java/org/xrpl/xrpl4j/model/flags/Flags.java

```diff
@@ -535,6 +536,289 @@ public boolean lsfOneOwnerCount() {
+      AccountSetTransactionFlags flags = new AccountSetTransactionFlags(value);
+
+      Preconditions.checkArgument(
+        !(flags.tfRequireDestTag() && flags.tfOptionalDestTag()),
+        "tfRequireDestTag and tfOptionalDestTag cannot both be set to true."
+      );
+
+      Preconditions.checkArgument(
+        !(flags.tfRequireAuth() && flags.tfOptionalAuth()),
+        "tfRequireAuth and tfOptionalAuth cannot both be set to true."
+      );
+
+      Preconditions.checkArgument(
+        !(flags.tfDisallowXrp() && flags.tfAllowXrp()),
+        "tfDisallowXrp and tfAllowXrp cannot both be set to true."
+      );
+
+      return flags;
+    }
+
+    /**
+     * Create a new {@link AccountSetTransactionFlags.Builder}.
+     *
+     * @return A new {@link AccountSetTransactionFlags.Builder}.
+     */
+    public static AccountSetTransactionFlags.Builder builder() {
+      return new AccountSetTransactionFlags.Builder();
+    }
+
+    /**
+     * Require a fully canonical signature.
+     *
+     * @return {@code true} if {@code tfFullyCanonicalSig} is set, otherwise {@code false}.
+     */
+    public boolean tfFullyCanonicalSig() {
+      return this.isSet(TransactionFlags.FULLY_CANONICAL_SIG);
+    }
+
+    /**
+     * Whether or not the {@code tfRequireDestTag} flag is set.
+     *
+     * @return {@code true} if {@code tfRequireDestTag} is set, otherwise {@code false}.
+     */
+    public boolean tfRequireDestTag() {
+      return this.isSet(REQUIRE_DEST_TAG);
+    }
+
+    /**
+     * Whether or not the {@code tfOptionalDestTag} flag is set.
+     *
+     * @return {@code true} if {@code tfOptionalDestTag} is set, otherwise {@code false}.
+     */
+    public boolean tfOptionalDestTag() {
+      return this.isSet(OPTIONAL_DEST_TAG);
+    }
+
+    /**
+     * Whether or not the {@code tfRequireAuth} flag is set.
+     *
+     * @return {@code true} if {@code tfRequireAuth} is set, otherwise {@code false}.
+     */
+    public boolean tfRequireAuth() {
+      return this.isSet(REQUIRE_AUTH);
+    }
+
+    /**
+     * Whether or not the {@code tfOptionalAuth} flag is set.
+     *
+     * @return {@code true} if {@code tfOptionalAuth} is set, otherwise {@code false}.
+     */
+    public boolean tfOptionalAuth() {
+      return this.isSet(OPTIONAL_AUTH);
+    }
+
+    /**
+     * Whether or not the {@code tfDisallowXrp} flag is set.
+     *
+     * @return {@code true} if {@code tfDisallowXrp} is set, otherwise {@code false}.
+     */
+    public boolean tfDisallowXrp() {
+      return this.isSet(DISALLOW_XRP);
+    }
+
+    /**
+     * Whether or not the {@code tfAllowXrp} flag is set.
+     *
+     * @return {@code true} if {@code tfAllowXrp} is set, otherwise {@code false}.
+     */
+    public boolean tfAllowXrp() {
+      return this.isSet(ALLOW_XRP);
+    }
+
+    /**
+     * A builder class for {@link AccountSetTransactionFlags}.
+     */
+    public static class Builder {
+      private boolean tfFullyCanonicalSig = true;
```

Comment:
Any reason for `private`, but only here and not the other fields?

Reply:
copy paste error :shame: